src/ripple/overlay/Overlay.h

```diff
@@ -174,73 +172,22 @@ class Overlay
     relay (protocol::TMValidation& m,
         uint256 const& uid) = 0;
 
-    /** Visit every active peer and return a value
-        The functor must:
-        - Be callable as:
-            void operator()(std::shared_ptr<Peer> const& peer);
-         - Must have the following type alias:
-            using return_type = void;
-         - Be callable as:
-            Function::return_type operator()() const;
-
-        @param f the functor to call with every peer
-        @returns `f()`
-
-        @note The functor is passed by value!
-    */
-    template <typename UnaryFunc>
-    std::enable_if_t<! std::is_void<
-            typename UnaryFunc::return_type>::value,
-                typename UnaryFunc::return_type>
-    foreach (UnaryFunc f)
-    {
-        for (auto const& p : getActivePeers())
-            f (p);
-        return f();
-    }
-
-    /** Visit every active peer
-        The visitor functor must:
-         - Be callable as:
-            void operator()(std::shared_ptr<Peer> const& peer);
-         - Must have the following type alias:
-            using return_type = void;
-
-        @param f the functor to call with every peer
-    */
+    /** Visit every active peer.
+     *
+     * The visitor must be invocable as:
+     *     Function(std::shared_ptr<Peer> const& peer);
+     *
+     * @param f the invocable to call with every peer
+     */
     template <class Function>
-    std::enable_if_t <
-        std::is_void <typename Function::return_type>::value,
-        typename Function::return_type
-    >
-    foreach(Function f)
+    // REVIEWER: Should we use enable_if in place of a concept for now?
```

Comment:
I’d prefer a constraint via `std::enable_if`, personally but up to you. Either way please either remove this comment; if you want to leave it please change it to a TODO.

Reply:
I removed the comment. There's only one overload now. A signature incompatibility will be diagnosed by a substitution error instead of a missing declaration error.